CMakeLists.txt

```diff
@@ -20,9 +20,35 @@ set(CMAKE_CXX_EXTENSIONS OFF)
 set(CMAKE_EXPORT_COMPILE_COMMANDS YES)
 
 if("${CMAKE_CXX_COMPILER_ID}" STREQUAL "Clang")
-  add_compile_options(-stdlib=libstdc++)
+  if(APPLE)
+    # Clang on macOS must use libc++
+    add_compile_options(-stdlib=libc++)
+  elseif(UNIX AND NOT APPLE)
+    # Clang on Linux can use libstdc++ (usually default) No need to explicitly
+    # add this unless required
+    add_compile_options(-stdlib=libstdc++)
+  endif()
+endif()
+
+# if(APPLE) execute_process( COMMAND xcrun --show-sdk-path OUTPUT_VARIABLE
+# MACOS_SDK_PATH OUTPUT_STRIP_TRAILING_WHITESPACE )
+# include_directories("${MACOS_SDK_PATH}/usr/include/c++/v1") endif()
+
+if(APPLE)
+  if(NOT DEFINED CMAKE_OSX_ARCHITECTURES)
+    set(CMAKE_OSX_ARCHITECTURES
+        "arm64"
+        CACHE STRING "" FORCE
+    )
+  endif()
 endif()
 
+if(APPLE)
+  set(Python3_ROOT_DIR "/opt/homebrew") # cmake-lint: disable=C0103
```

Comment:
Oh boy, someone should fix the cmake find package for python :laughing: or why is it necessary to specify the actual root beforehand where it's not on other systems?